Add TRIAGING.md for triage agenda and processes (#11814)
opensearch-project/OpenSearch: TRIAGING.md

<img src="https://opensearch.org/assets/img/opensearch-logo-themed.svg" height="64px">

The maintainers of the OpenSearch Repo seek to promote an inclusive and engaged community of contributors. In order to facilitate this, weekly triage meetings are open-to-all and attendance is encouraged for anyone who hopes to contribute, discuss an issue, or learn more about the project. To learn more about contributing to the OpenSearch Repo visit the [Contributing](./CONTRIBUTING.md) documentation.

### Do I need to attend for my issue to be addressed/triaged?

Attendance is not required for your issue to be triaged or addressed. All new issues are triaged weekly.

> 

Reviewer: I would add something like "Any actionable results of discussion of the issue will be a comment in the issue."

### What happens if my issue does not get covered this time?

Each meeting we seek to address all new issues. However, should we run out of time before your issue is discussed, you are always welcome to attend the next meeting or to follow up on the issue post itself.

### How do I join the Triage meeting?

> Meetings are hosted regularly at 10:00a - 10:55a Central Time every Wednesday and can be joined via [Chime](https://chime.aws/1988437365).

Author: Picking any time is going to include some and exclude others - I'm willing to be flexible if I can get some others to commit with me to keep this up weekly

After joining the Chime meeting, you can enable your video / voice to join the discussion.  If you do not have a webcam or microphone available, you can still join in via the text chat.

If you have an issue you'd like to bring forth please prepare a link to the issue so it can be presented and viewed by everyone in the meeting.

### Is there an agenda for each week?

Meetings are 55 minutes and structured as follows:

1. Initial Gathering: As we gather, feel free to turn on video and engage in informal and open-to-all conversation.  After a bit a volunteer will share their screen and proceed with the agenda.

> 2. Collect attendee annotations: The meeting driver will ask all triage attendee to post a link to their GitHub profile, these will be collected and assembled these into a signature to annotation all issues triaged during this meeting.

Reviewer 2: ```suggestion
2. Collect attendee annotations: The meeting driver will ask all triage attendees to post a link to their GitHub profile, these will be collected and assembled into a signature to annotated all issues triaged during this meeting.
```

I think the initial sentence here needs a bit of restructuring, suggesting the favour but not sure I captured the intent right.

Author: Thanks - I'll wordsmith this section more too.

   - Format of annotation should follow ``[Triage - attendees [1](https://github.com/peternied), [2](...)]``
3. Announcements: If there are any announcements to be made they will happen at the start of the meeting.
4. Review of New Issues: The meetings always start with reviewing all untriaged [issues](https://github.com/search?q=label%3Auntriaged+is%3Aopen++repo%3Aopensearch-project%2FOpenSearch+&type=issues&ref=advsearch&s=created&o=desc) for the OpenSearch repo.

> 5. Member Requests: Opportunity for any meeting member to ask for consideration of an issue or pull request.

Reviewer 2: ```suggestion
5. Attendee Requests: Opportunity for any meeting attendee to ask for consideration of an issue or pull request.
```

It think we use attendee in other places, may be we could stick to that?

> 6. Open Discussion: Allow for members of the meeting to surface any topics without issues filed or pull request created.

Reviewer 2: ```suggestion
6. Open Discussion: Allow for attendees of the meeting to surface any topics without issues filed or pull request created.
```

### What are the issue labels associated with triaging?

Yes, there are several labels that are used to identify the 'state' of issues filed in OpenSearch .
| Label | When applied | Meaning |
| ----- | ------------ | ------- |
| Untriaged | When issues are created or re-opened. | Issues labeled as 'Untriaged' require the attention of the repository maintainers and may need to be prioritized for quicker resolution. It's crucial to keep the count of 'Untriaged' labels low to ensure all potential security issues are addressed in a timely manner. See [SECURITY.md](https://github.com/opensearch-project/OpenSearch/blob/main/SECURITY.md) for more details on handling these issues. |
| Help Wanted | Anytime. | Issues marked as 'Help Wanted' signal that they are actionable and not the current focus of the project maintainers. Community contributions are especially encouraged for these issues. |
| Good First Issue | Anytime. | Issues labeled as 'Good First Issue' are small in scope and can be resolved with a single pull request. These are recommended starting points for newcomers looking to make their first contributions. |

### What kind of outcomes will a triaged issue have?

> | Outcome | Label | Description | Canned Response | 

Author: I plan on having the triaging.md page up when running triage and liberally copy/pasting from these canned responses for a relatively consistent experience even if the meeting drivers change.  Anyone have experience doing this or thoughts about improving this approach?

Reviewer: this is pretty good. so, it's not specifically the outcome for each issue, but a way to build up templates without having to dive into each issue to find something similar?

Author: Yup - a time optimization and helps reduce the mental resources need by the triage meeting.  If the outcome is already on the list, next steps are easy.  If we run into outcomes that aren't on this list, there might be a meta issue to document another outcome on this list.

| ------- | ----- | ----------- | --------------- |
| Accepted | -`untriaged` | The issue has the details needed to be directed towards area owners. | "Thanks for filing this issue, please feel free to submit a pull request." |
| Rejected | N/A | The issue will be closed with a reason for why it was rejected.  Some reasons will be due to lack of details, or being outside the scope of the project. | "Thanks for creating this issue; however, it isn't being accepted due to {REASON}.  Please feel free to re-open after addressing the reason." |

> | Area Triage | +`{AREA-LABEL}` | OpenSearch has many different areas and knowing if an issue should be accepted or not might not be possible with the representatives in the triage meeting, it will be labeled with the area and an owner will be @mentioned to follow up | "Thanks for creating this issue; the triage meeting was unsure if this issue should be accepted, @{PERSON} or someone from the area please review and then accept or reject this issue?" |

Author: Area owners - what do you think of this?  IMO this is a low friction way to delegate out deep dives or subject matter experts to consult on issues in areas they own.

Reviewer: Maybe this is a good use case for teams? each area has a team and each team figures out how they're going to address the "Area Triage."

Author: opensearch-project's organization does not allow non-amazon employees, which means that none of contributors can be included in these org based groups.  :'(

I'd love to use teams if we could include everyone.

<img src="https://opensearch.org/assets/img/opensearch-logo-themed.svg" height="64px">

The maintainers of the OpenSearch Repo seek to promote an inclusive and engaged community of contributors. In order to facilitate this, weekly triage meetings are open-to-all and attendance is encouraged for anyone who hopes to contribute, discuss an issue, or learn more about the project. To learn more about contributing to the OpenSearch Repo visit the [Contributing](./CONTRIBUTING.md) documentation.

### Do I need to attend for my issue to be addressed/triaged?

Attendance is not required for your issue to be triaged or addressed.  If not accepted the issue will be updated with a comment for next steps.  All new issues are triaged weekly.

You can track if your issue was triaged by watching your GitHub notifications for updates.

### What happens if my issue does not get covered this time?

Each meeting we seek to address all new issues. However, should we run out of time before your issue is discussed, you are always welcome to attend the next meeting or to follow up on the issue post itself.

> ### How do I join the Triage meeting?

Reviewer 3: How this can be scaled? 1 hour might not be enough to triage all incoming issues in one week. Are we planning to increase the frequency?

Author: We will have to run a couple and see what needs to be tweaked, meeting duration/frequency could have a big impact.

### What if I have an issue that is almost a duplicate, should I open a new one to be triaged?

You can always open an [issue](https://github.com/opensearch-project/OpenSearch/issues/new/choose) including one that you think may be a duplicate. If you believe your issue is similar but distinct from an existing one, you are encouraged to file it and explain the differences during the triage meeting.

### What if I have follow-up questions on an issue?

If you have an existing issue you would like to discuss, you can always comment on the issue itself. Alternatively, you are welcome to come to the triage meeting to discuss.

### Is this meeting a good place to get help setting up features on my OpenSearch instance?

While we are always happy to help the community, the best resource for implementation questions is [the OpenSearch forum](https://forum.opensearch.org/).

There you can find answers to many common questions as well as speak with implementation experts.

> ### What are the issue labels associated with triaging?

Reviewer 3: I think after having component's label on issue creation, Are you still targeting to triage all issues, or only issues that has both untriaged and other labels together. Because in other cases, the issue will be labled with a component/area and the owner of that area will follow up.

Author: This will be a triage for all issue in OpenSearch - the queries links are in the agenda.  If we have a security issue filed in an area; finding it quickly is important so it can be addresses as soon as possible, this weekly triage is targeting keeping that  max time to 7 days.
